sound/soc/sof/ipc4-topology.c

```diff
@@ -1419,13 +1419,55 @@ static int sof_ipc4_control_load_volume(struct snd_sof_dev *sdev, struct snd_sof
 	return 0;
 }
 
+static int sof_ipc4_control_load_bytes(struct snd_sof_dev *sdev, struct snd_sof_control *scontrol)
+{
+	struct sof_ipc4_control_data *control_data;
+	int ret;
+
+	scontrol->size = sizeof(struct sof_ipc4_control_data) + scontrol->priv_size;
+
+	scontrol->ipc_control_data = kzalloc(scontrol->size, GFP_KERNEL);
+	if (!scontrol->ipc_control_data)
+		return -ENOMEM;
+
+	control_data = scontrol->ipc_control_data;
+	control_data->index = scontrol->index;
+	if (scontrol->priv_size > 0) {
+		memcpy(control_data->data, scontrol->priv, scontrol->priv_size);
+
```

Comment:
scontrol->priv can be freed at this point?

Reply:
scontrol->priv is freed in sof_control_unload() in topology.c. For current code, it is safe to free scontrol->priv. But I'm not sure whether someone will submit other feature patches which use scontrol->priv later. Anyway, I will add the free code in next update.

Reply:
If scontrol->priv can be freed here, I think it should be freed at every error above?